beacon-chain/blockchain/stategenerator/state_generator.go

```diff
@@ -67,12 +78,32 @@ func GenerateStateFromSlot(ctx context.Context, db *db.BeaconDB, slot uint64) (*
 		if err != nil {
 			return nil, fmt.Errorf("could not execute state transition %v", err)
 		}
+	}
+
+	log.Debugf("End: Current slot %d and Finalized Epoch %d", hState.Slot, hState.FinalizedEpoch)
+
+	return hState, nil
+}
+
+// ancestersToLastFinalizedBlock will return a list of all block ancesters
+// between the given block and the most recent finalized block in the db.
+// The given block is also returned in the list of ancesters.
+func ancestersToLastFinalizedBlock(db *db.BeaconDB, block *pb.BeaconBlock,
+	finalizedBlockRoot [32]byte) ([]*pb.BeaconBlock, error) {
+
+	blockAncestors := make([]*pb.BeaconBlock, 0)
```

Comment:
If you passed in the finalized block, you could access the slot. You would then know that the max number of ancestors would be block.Slot-finalizedBlock.Slot